src/javascript/binary/websocket_pages/user/new_account/virtual_acc_opening.js

```diff
@@ -1,92 +1,115 @@
+        $('#email_consent').parent().parent().removeClass('invisible');
     };
 
-    const init = function() {
-        Content.populate();
-        handleResidence();
-        BinarySocket.send({ residence_list: 1 });
-        BinarySocket.send({ website_status: 1 });
+    const populateRequest = () => {
+        const req = $.extend({ new_account_virtual: 1 }, getFormData());
 
-        const form = $('#virtual-form')[0];
-        if (!form) return;
+        // Add TrafficSource parameters
+        const utm_data = TrafficSource.getData();
+        req.utm_source = TrafficSource.getSource(utm_data);
+        if (utm_data.utm_medium)   req.utm_medium   = utm_data.utm_medium;
+        if (utm_data.utm_campaign) req.utm_campaign = utm_data.utm_campaign;
 
-        bind_validation.simple(form, {
-            schema: VirtualAccOpeningData.getSchema(),
-            submit: function(ev, info) {
-                ev.preventDefault();
-                if (info.errors.length > 0) {
-                    return;
-                }
-                configureSocket();
-                const data = info.values;
-                VirtualAccOpeningData.newAccount({
-                    password         : data.password,
-                    residence        : (japanese_client() ? 'jp' : data.residence),
-                    verification_code: data['verification-code'],
-                });
-            },
-        });
+        const gclid = Client.get('gclid');
+        if (gclid) req.gclid_url = gclid;
+
+        if (Cookies.get('affiliate_tracking')) {
+            req.affiliate_token = Cookies.getJSON('affiliate_tracking').t;
+        }
+        return req;
     };
 
-    const onLoad = function() {
-        if (Client.is_logged_in()) {
-            window.location.href = url_for('home');
-            return;
+    const handleNewAccount = (response) => {
+        if (!response) return false;
+        const error = response.error;
+        if (!error) {
+            const new_account = response.new_account_virtual;
+            Client.set_cookie('residence', response.echo_req.residence);
+            return Client.process_new_account(
+                new_account.email,
+                new_account.client_id,
+                new_account.oauth_token,
+                true);
+        }
+
+        switch (error.code) {
+            case 'InvalidToken': {
+                const message = 'Your token has expired. Please click <a class="pjaxload" href="[_1]">here</a> to restart the verification process.';
+                return showFormError(message, '');
+            }
+            case 'duplicate email': {
+                const message = 'Your provided email address is already in use by another Login ID. According to our terms and conditions, you may only register once through our site. If you have forgotten the password of your existing account, please <a href="[_1]">try our password recovery tool</a> or contact customer service.';
```

Comment:
@negarn just a suggestion: 'The email address provided is already in use. If you forgot your password, please try our <a href="[_1]">password recovery tool</a> or contact our <a href="">customer service</a>.'